gems/aws-sdk-core/lib/aws-sdk-core/plugins/api_key.rb

```diff
@@ -21,7 +21,7 @@ def add_handlers(handlers, config)
       class OptionHandler < Seahorse::Client::Handler
         def call(context)
           # apply APIG user-agent by default
```

Comment:
I think either here or in the near future we should pull this out of the API Key plugin - this wouldn't cause harm if the API Key plugin were removed, but it's turning this plugin into a catch-all for bringing along other configuration.